test/burnside/expr-test-generator.cc

```diff
@@ -1,17 +1,3 @@
-// Copyright (c) 2018-2019, NVIDIA CORPORATION.  All rights reserved.
-//
-// Licensed under the Apache License, Version 2.0 (the "License");
-// you may not use this file except in compliance with the License.
-// You may obtain a copy of the License at
-//
-//     http://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
-
```

Comment:
This looks odd. Why wasn't a new header inserted?

Reply:
The LLVM style is to not include headers at all in test files. They show up only rarely in LLVM itself.

Reply:
OK. This is a oddball case, as it's a program to generate test(s) in the test area, as opposed to a bunch of test inputs that are meant to be processed by test tools. Which makes me wonder if this file belongs in some other place like utils or tools.

Reply:
I think I can easily leave it as having the header if you like.

Reply:
That's ok with me. I'll ping Jean since this came from him...

Reply:
I wrote this code from scratch so I guess you can licence it in whatever way which is OK with LLVM/Nvidia licensing practices.
I agree with Eric that this is a test tool more than a test file. I can see that MLIR has LLVM licences in test infrastructure related code (example here: https://github.com/llvm/llvm-project/blob/master/mlir/test/lib/IR/TestFunc.cpp) as opposed to test files (like here: https://github.com/llvm/llvm-project/blob/master/mlir/test/AffineOps/canonicalize.mlir). So I would keep a licence here, but I am also OK if you think it is weird/unnecessary/time-consuming.